torch/csrc/jit/python_compiled_function.cpp

```diff
@@ -90,18 +97,37 @@ struct CompiledFunction {
       } catch(const jit::NotImplementedException & ex) {
         closure_ = std::make_shared<AutogradClosureFactory>(complete_trace.get());
       }
+      prepareStageDetails(complete_trace.get());
+      graph_ = complete_trace->graph;
       is_ready_ = true;
       return true;
     }
 
+    void prepareStageDetails(TracingState *state) {
```

Comment:
python_compiled_function.cpp is probably not the right place for this. We probably need an object like AutogradClosureFactory that is constructed with a TracingState and then produces AutogradInterpreterFunctions. This wasn't needed before because the only argument to AutogradInterpreterFunction was jit::Code.